src/MessageInterface.php

```diff
@@ -129,7 +129,7 @@ public function getHeaderLine(string $name): string;
      * @return static
      * @throws \InvalidArgumentException for invalid header names or values.
      */
-    public function withHeader(string $name, $value): static;
+    public function withHeader(string $name, array|string $value): static;
```

Comment:
Thinking about it, this should rather accept `mixed` to include scalar types, as some implementations are [already implemented this way](https://github.com/guzzle/psr7/blob/815698d9f11c908bc59471d11f642264b533346a/tests/ResponseTest.php#L169-L173), ignoring the defined type in the docblock.